### PR TITLE
Bumps jetty version to 9.4.52.v20230823 to fix GMS-2023-1857

### PR DESCRIPTION
**Pull request by DarshitChanpura**
Fixes GMS-2023-1857.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kkmr — APPROVED** [review]

File: test/fixtures/hdfs-fixture/build.gradle [hunk 1]
```diff
@@ -33,7 +33,7 @@ apply plugin: 'opensearch.java'
 group = 'hdfs'
 
 versions << [
-  'jetty': '9.4.51.v20230217'
+  'jetty': '12.0.1'
```
Comment (reta) [inline test/fixtures/hdfs-fixture/build.gradle:36]: @DarshitChanpura this is a stretch probably, AFAIK 9.x and 12.x are not compatible

Comment (reta) [reply test/fixtures/hdfs-fixture/build.gradle]: We may need to use 9.4.52.v20230823, it fixes GMS-2023-1857 as well, as per release notes https://github.com/eclipse/jetty.project/releases/tag/jetty-9.4.52.v20230823

Comment (DarshitChanpura) [reply test/fixtures/hdfs-fixture/build.gradle]: hmm. I updated them separately.

Comment (reta) [reply test/fixtures/hdfs-fixture/build.gradle]: Why? Should be just

```
'jetty': '9.4.52.v20230823'
```

I think

Comment (DarshitChanpura) [reply test/fixtures/hdfs-fixture/build.gradle]: done. I went with a basic thinking to upgrade both to their latest versions but that seems to cause:
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/eclipse/jetty/util/MultiException
```

i have synced both to `9.4.52.v20230823`

**Review by reta — APPROVED** [review]

**Comment by DarshitChanpura** [comment]
test failures seem to be flaky.

**Comment by reta** [comment]
@DarshitChanpura could you please rebase? I think those were fixed on `main`

**Comment by reta** [comment]
@DarshitChanpura could you please manually backport to `2.x`? thank you

**Pull request merged**